src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -2179,220 +2188,1098 @@ void ScriptMgr::OnBeforeMailDraftSendMailTo(MailDraft* mailDraft, MailReceiver c
     FOREACH_SCRIPT(MailScript)->OnBeforeMailDraftSendMailTo(mailDraft, receiver, sender, checked, deliver_delay, custom_expiration, deleteMailItemsFromDB, sendMail);
 }
 
-AllMapScript::AllMapScript(const char* name)
-    : ScriptObject(name)
+void ScriptMgr::OnBeforeUpdatingPersonalRating(int32& mod, uint32 type)
 {
-    ScriptRegistry<AllMapScript>::AddScript(this);
+    FOREACH_SCRIPT(FormulaScript)->OnBeforeUpdatingPersonalRating(mod, type);
 }
 
-AllCreatureScript::AllCreatureScript(const char* name)
-    : ScriptObject(name)
+bool ScriptMgr::OnBeforePlayerQuestComplete(Player* player, uint32 quest_id)
 {
-    ScriptRegistry<AllCreatureScript>::AddScript(this);
+    bool ret=true;
+    FOR_SCRIPTS_RET(PlayerScript, itr, end, ret) // return true by default if not scripts
+    if (!itr->second->OnBeforeQuestComplete(player, quest_id))
+        ret=false; // we change ret value only when scripts return false
```

Comment:
are we returnig `false` if the LAST value of `FOR_SCRIPTS_RET` is false or if ANY of them returns false?